botocore/loaders.py

```diff
@@ -355,7 +372,32 @@ def load_service_model(self, service_name, type_name, api_version=None):
             api_version = self.determine_latest_version(
                 service_name, type_name)
         full_path = os.path.join(service_name, api_version, type_name)
-        return self.load_data(full_path)
+        model = self.load_data(full_path)
+
+        if not load_extras:
+            return model
+
+        # Load in all the extras
+        for tool_name in self._tool_extras:
+            self._load_extras(
+                model, service_name, type_name, api_version, tool_name)
+
+        return model
+
+    def _load_extras(self, service_model, service_name, type_name,
```

Comment:
This doesn't seem like the right verb.  In the loaders, "load" is actually finding something on disk and deserializing it.  Here we'll really merging/updating an existing model.  I'd prefer to have that be named distinctly from "load".  Something like `self._merge_extras`.